### PR TITLE
[zero] get memory usage for sharded param

### PR DESCRIPTION
**Pull request by feifeibear**
A PR towards tracing non-model data memory usage inside an iteration.

### CODE REVIEW

File: colossalai/zero/sharded_param/sharded_param.py [hunk 1]
```diff
@@ -40,3 +40,28 @@ def sharded_data_tensor(self):
...
+        cuda_mem_use, cpu_mem_use = 0, 0
+
+        def _update_mem_use(t: Optional[torch.Tensor]):
+            if t is None:
+                return
+            assert isinstance(t, torch.Tensor)
+            nonlocal cuda_mem_use
+            nonlocal cpu_mem_use
+            if t.device.type == 'cpu':
+                cpu_mem_use += t.numel() * t.element_size()
```
Comment (Gy-Lu) [inline colossalai/zero/sharded_param/sharded_param.py:59]: Seems it would return the number of Byte, do you consider formatting it to KB/MB/GB?

Comment (feifeibear) [reply colossalai/zero/sharded_param/sharded_param.py:59]: Yeah, we can convert the unit at the time it is used. Currently, we can just make the API as simple as possible.

Comment (Gy-Lu) [reply colossalai/zero/sharded_param/sharded_param.py:59]: ok

**Review by Gy-Lu — APPROVED** [review]

File: colossalai/zero/sharded_param/sharded_param.py [hunk 2]
```diff
@@ -40,3 +40,28 @@ def sharded_data_tensor(self):
...
+            elif t.device.type == 'cuda':
+                cuda_mem_use += t.numel() * t.element_size()
+
+        _update_mem_use(self.sharded_data_tensor.payload)
```
Comment (1SAA) [inline colossalai/zero/sharded_param/sharded_param.py:63]: If the tensor is shared but it payload hasn't been removed from CUDA,
I think we should caculate the memory of param.data in this case.

**Pull request merged**